client/src/components/Panels/Main/Editor/Editor.tsx

```diff
@@ -252,51 +252,108 @@ const Editor = () => {
     //eslint-disable-next-line react-hooks/exhaustive-deps
   }, [editor, explorer, explorerChanged, setNoOpenTabs]);
 
+  const updateLibRs = (currentContent: string, programId: string) => {
```

Comment:
`updateLibRs` and `updateLibPy` seems to be nearly identical functions. Would be nice to have a single function with an identifier argument.

Reply:
Good shout, I've refactored this to a single function. Definitely should be easier to maintain now!